### PR TITLE
Revamp dotnet-pgo arguments

### PR DESCRIPTION
**Pull request by davidwrighton**
- Add a document describing the behavior and how it is expected to be used.
- Rework the existing command line handling to implement the proposed create-mibc command. (the merge command is still entirely TBD).

### CODE REVIEW

**Comment by davidwrighton** [comment]
@dotnet/crossgen-contrib

@AndyAyersMS @briansull Andy, could you take a look at the spec I wrote up, and provide feedback?

File: docs/design/features/dotnet-pgo.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+$ dotnet tool install --global dotnet-pgo
+You can invoke the tool using the following command: dotnet-pgo
+Tool 'dotnet-pgo' (version '6.0.47001') was successfully installed.
+```
+
+## Using dotnet-pgo to optimize an application
+
+In order to use dotnet-pgo you will need to:
+
+1. Collect traces of you application with at least the following trace setting. `Microsoft-Windows-DotNETRuntime:0x1C000080018:4`
```
Comment (AndyAyersMS) [inline docs/design/features/dotnet-pgo.md:24]: Maybe an example of doing this, say via `dotnet-trace`?

File: docs/design/features/dotnet-pgo.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    --compressed                     Generate compressed mibc
+    -h, --help                       Display this usage message.
+```
+
+`dotnet-pgo create-mibc` is used to create mibc file from a trace. It supports nettrace, etl, netperf, and netperf.zip trace file formats. The `--trace-file` and `--output-file-name` switches must be specified.
+
+If using trace file format that supports multiple process tracing, specify some combination of the `--pid`, `--process-name` and `--clr-instance-id` switches as needed. When run from the command line in situations where these arguments are needed, the valid args will be printed to the console. Usage of the nettrace file format is recommended for users attempting to run this tool as part of a CI system.
+
+The `dotnet-pgo` tool is designed to run on a machine which has the same file system layout as the trace. If this is not the case, `dotnet-pgo` may not be able to find assemblies loaded by the application. In those cases, the `--reference` switch may be used to inform `dotnet-pgo` where to find assemblies that are part of the application.
+
```
Comment (AndyAyersMS) [inline docs/design/features/dotnet-pgo.md:63]: Is the argument for `--reference` supposed to be a full path to an assembly, or a directory where the tool will search for missing assemblies, or ...?

File: docs/design/features/dotnet-pgo.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+```
+usage: dotnet-pgo merge --input <arg>... --output-file-name <arg> [--exclude-reference <arg>] [--verbosity <arg>] [--compressed] [-h]
+
+    --input <arg>                    Specify the trace file to be parsed.
+    --output-file-name <arg>         Specify the output filename to be created.
+    --exclude-reference <arg>        Exclude references to the specified assembly from the output.
+    --verbosity <arg>                Adjust verbosity level.
+    --compressed                     Generate compressed mibc
+    -h, --help                       Display this usage message.
+```
```
Comment (AndyAyersMS) [inline docs/design/features/dotnet-pgo.md:80]: This is likely going to need some sort of weighting/prioritization capability. Without it the longer-running scenario profile data may dominate the merged results.

And to figure out suitable weights we may need some sort of analysis capability....

File: docs/design/features/dotnet-pgo.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+usage: dotnet-pgo create-mibc --trace-file <arg> --output-file-name <arg> [--pid <arg>] [--process-name [arg]] [--clr-instance-id <arg>] [--reference <arg>...] [--exclude-events-before <arg>]
+                  [--exclude-events-after <arg>] [--verbosity <arg>] [--compressed] [-h]
+
+    --trace-file <arg>               Specify the trace file to be parsed.
+    --output-file-name <arg>         Specify the output filename to be created.
+    --pid <arg>                      The pid within the trace of the process to examine. If this is a multi-process trace, at least one of --pid or --process-name must be specified
+    --process-name [arg]             The process name within the trace of the process to examine. If this is a multi-process trace, at least one of --pid or --process-name must be specified.
+    --clr-instance-id <arg>          If the process contains multiple .NET runtimes, the instance ID must be specified.
+    --reference <arg>...             If a reference is not located on disk at the same location as used in the process, it may be specified with a --reference parameter
```
Comment (davidwrighton) [inline docs/design/features/dotnet-pgo.md:50]: Support -r in addition to --reference

File: docs/design/features/dotnet-pgo.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+### `create-mibc` command
+Transform a trace file into a Mibc profile data file.
+```
+usage: dotnet-pgo create-mibc --trace-file <arg> --output-file-name <arg> [--pid <arg>] [--process-name [arg]] [--clr-instance-id <arg>] [--reference <arg>...] [--exclude-events-before <arg>]
+                  [--exclude-events-after <arg>] [--verbosity <arg>] [--compressed] [-h]
+
+    --trace-file <arg>               Specify the trace file to be parsed.
+    --output-file-name <arg>         Specify the output filename to be created.
```
Comment (davidwrighton) [inline docs/design/features/dotnet-pgo.md:46]: Rename to --output and also support -o to be more similar to the dotnet-trace tool.

File: docs/design/features/dotnet-pgo.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    merge          Merge multiple Mibc profile data files into one file.
+```
+
+### `create-mibc` command
+Transform a trace file into a Mibc profile data file.
+```
+usage: dotnet-pgo create-mibc --trace-file <arg> --output-file-name <arg> [--pid <arg>] [--process-name [arg]] [--clr-instance-id <arg>] [--reference <arg>...] [--exclude-events-before <arg>]
+                  [--exclude-events-after <arg>] [--verbosity <arg>] [--compressed] [-h]
+
+    --trace-file <arg>               Specify the trace file to be parsed.
```
Comment (davidwrighton) [inline docs/design/features/dotnet-pgo.md:45]: Rename to `--trace` and support `-t` to reduce verbosity.

File: docs/design/features/dotnet-pgo.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+The mibc file format may be compressed. This will require slightly more time to create and process, but it will produce a smaller binary if the mibc files needs to be distributed to multiple developers. Create a compressed mibc by using the `--compressed` switch.
+
+### `merge` command
+Merge multiple Mibc profile data files into one file.
+
+```
+usage: dotnet-pgo merge --input <arg>... --output-file-name <arg> [--exclude-reference <arg>] [--verbosity <arg>] [--compressed] [-h]
+
+    --input <arg>                    Specify the trace file to be parsed.
+    --output-file-name <arg>         Specify the output filename to be created.
```
Comment (davidwrighton) [inline docs/design/features/dotnet-pgo.md:75]: Rename to `--output` and also support `-o`

File: docs/design/features/dotnet-pgo.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+The mibc file format may be compressed. This will require slightly more time to create and process, but it will produce a smaller binary if the mibc files needs to be distributed to multiple developers. Create a compressed mibc by using the `--compressed` switch.
+
+### `merge` command
+Merge multiple Mibc profile data files into one file.
+
+```
+usage: dotnet-pgo merge --input <arg>... --output-file-name <arg> [--exclude-reference <arg>] [--verbosity <arg>] [--compressed] [-h]
+
+    --input <arg>                    Specify the trace file to be parsed.
```
Comment (davidwrighton) [inline docs/design/features/dotnet-pgo.md:74]: Add support for `-i` short form

File: docs/design/features/dotnet-pgo.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    --output-file-name <arg>         Specify the output filename to be created.
+    --exclude-reference <arg>        Exclude references to the specified assembly from the output.
+    --verbosity <arg>                Adjust verbosity level.
```
Comment (davidwrighton) [inline docs/design/features/dotnet-pgo.md:77]: Add support for `-v` and document that the supported levels are minimal, normal, detailed, and diagnostic.

File: docs/design/features/dotnet-pgo.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+    --trace-file <arg>               Specify the trace file to be parsed.
+    --output-file-name <arg>         Specify the output filename to be created.
+    --pid <arg>                      The pid within the trace of the process to examine. If this is a multi-process trace, at least one of --pid or --process-name must be specified
+    --process-name [arg]             The process name within the trace of the process to examine. If this is a multi-process trace, at least one of --pid or --process-name must be specified.
+    --clr-instance-id <arg>          If the process contains multiple .NET runtimes, the instance ID must be specified.
+    --reference <arg>...             If a reference is not located on disk at the same location as used in the process, it may be specified with a --reference parameter
+    --exclude-events-before <arg>    Exclude data from events before specified time. Time is specified as milliseconds from the start of the trace.
+    --exclude-events-after <arg>     Exclude data from events after specified time. Time is specified as milliseconds from the start of the trace.
+    --verbosity <arg>                Adjust verbosity level.
```
Comment (davidwrighton) [inline docs/design/features/dotnet-pgo.md:53]: Add support for `-v` short form and document the supported levels (minimal, normal, detailed, diagnostic)

File: docs/design/features/dotnet-pgo.md (hunk 3 above)
Comment (davidwrighton) [reply docs/design/features/dotnet-pgo.md:80]: Agreed, but I think we can bolt that on to this by adding paremeters to the input by adding a ,[flags] after the filename on input.

File: docs/design/features/dotnet-pgo.md (hunk 2 above)
Comment (davidwrighton) [reply docs/design/features/dotnet-pgo.md:63]: This argument is a file or path with wild cards. I'll add a note.

**Review by davidwrighton — COMMENTED** [review]
Add comments based on attempting to make the naming a bit more consistent.

**Review by AndyAyersMS — APPROVED** [review]
LGTM.

**Pull request merged**